### PR TITLE
feature/chris-7355-revise-rx-filter-and-sort

### PR DESCRIPTION
**Pull request by alexandec**
## Description of Change

Overview of RX filter/sort changes in this PR:
- Introduce dynamic headers that follow the selected filter
- Consolidate filter/sort modals into a single modal
- Modify the modal to simplify the buttons, use the new Button component, change the filter and sort options, and show counts for each filter
- Change the default sort field
- Modify unit and e2e tests to reflect the above changes

This is the second of two PRs merging into the `feature/chris-rx-filter-improvements` branch. The first is #7534 which removed the RX tabs and reordered some components.

## Screenshots/Video

Before:

https://github.com/department-of-veterans-affairs/va-mobile-app/assets/5805150/c0440819-89b6-49ad-9326-2eb44ccf9383

After:

https://github.com/department-of-veterans-affairs/va-mobile-app/assets/5805150/010dd62f-0332-4559-b510-718a0a55e00c

## Testing

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations

1. Introduce dynamic list titles
- Dynamic list title
When All/Active/Discontinued/Expired/Transferred/Unknown filters are selected, the header will be: [Filter option] prescriptions (# of RX), sorted by [sort option]
When Pending filter is selected, the header will be: Pending refills (x of RX), sorted by [sort option]
When Tracking filter is selected, the header will be: Refills with tracking (x of RX), sorted by [sort option]
Sort option isn't capitalized in this header (i.e., ...sorted by status (A to Z), NOT ...sorted by Status (A to Z)
- Dynamic description under list title
When All/Active/Discontinued/Expired/Transferred/Unknown filters are selected, the description under the header will be: "This list only shows prescriptions filled by VA pharmacies and may not include all your medications."
When Pending filter is selected, the description will be: "This list shows refill requests you've submitted. It also shows refills the VA pharmacy is processing."
When Tracking filter is selected, the description will be: "This list shows refills with current tracking information available."

2. Revise current experience
 Remove current [Filter by / Sort by] buttons
 Remove [Sort by] panel
 Create "Filter and sort" Secondary Button
 Create a panel with a "Filter and sort" header
 Add Cancel - top left
 Add Apply - top right

3. Filter + Sort Panel
 - Create "Filter by" section
 All (with number of RX with this status)
 Active (with number of RX with this status)
 Apply Helper Text style: "Includes these statuses: On hold, Parked, Refill in process, and Submitted"
 Discontinued (with number of RX with this status)
 Expired (with number of RX with this status)
 Pending (with number of RX that fit this)
 Apply Helper Text style: "Includes refill requests you submitted and refills the VA pharmacy is processing"
 Tracking (with number of RX that fit this)
 Apply Helper Text style: "Includes refills with current tracking information available"
 Transferred (with number of RX with this status)
 Unknown (with number of RX with this status)

 - Create "Sort by" section
 Fill date (newest to oldest)
 Medication name (A to Z)
 Refills left (least to most)
 Status (A to Z)

- Enable Default states
Filter by = All
Sort by = Status (A to Z)

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [ ] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [ ] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

**Comment by alexandec** [comment]
@rbontrager if you could take a look at the E2E test changes here that would be great. I've updated everything so it passes with the new filter/sort approach, but just want to make sure I didn't miss anything that needs to be tested

File: VAMobile/src/components/FormWrapper/FormFields/RadioGroup.tsx [hunk 1]
```diff
@@ -128,9 +128,8 @@ const RadioGroup = <T,>({ options, value, onChange, disabled = false, error, isR
       const textLines: Array<TextLine> = [{ text: option.labelKey, variant: 'VASelector', color: disabled ? 'checkboxDisabled' : 'primary' }]
 
       if (option.additionalLabelText && option.additionalLabelText.length > 0) {
-        textLines[0].variant = 'MobileBodyBold'
         option.additionalLabelText.forEach((item) => {
-          textLines.push({ text: item, variant: 'MobileBody' })
+          textLines.push({ text: item, variant: 'HelperText' })
```
Comment (Sparowhawk) [inline VAMobile/src/components/FormWrapper/FormFields/RadioGroup.tsx:132]: Doesn't this affect the gender identity screen and mailing address validation screen? Is this change acceptable for those two screens?

**Review by Sparowhawk — COMMENTED** [review]
Just the one question on if it affects the mailing address validation and gender identity screens. If it doesn't then this is good to go. If it does then we probably need to ask design which is correct.

File: VAMobile/src/components/FormWrapper/FormFields/RadioGroup.tsx (hunk 1 above)
Comment (alexandec) [reply VAMobile/src/components/FormWrapper/FormFields/RadioGroup.tsx:132]: I searched the app for `additionalLabelText` and it's not used anywhere except in PrescriptionHistory, so this change shouldn't affect anything else in the app. The areas you mention use textLines, but they don't use additionalLabelText

**Review by Sparowhawk — APPROVED** [review]

**Comment by timwright12** [comment]
@alexandec @rbontrager I sent this back to code review because it looks like there's still an open ask to review the detox test updates

**Comment by alexandec** [comment]
@timwright12 should be fine as mentioned in [the ticket](https://github.com/department-of-veterans-affairs/va-mobile-app/issues/7355)

**Review by DJUltraTom — APPROVED** [review]

**Pull request merged**